llvm/lib/Target/AMDGPU/AMDGPUAtomicOptimizer.cpp

```diff
@@ -876,7 +876,7 @@ void AMDGPUAtomicOptimizerImpl::optimizeAtomic(Instruction &I,
   BasicBlock *Predecessor = nullptr;
   if (ValDivergent && ScanImpl == ScanOptions::Iterative) {
     // Move terminator from I's block to ComputeEnd block.
-    Instruction *Terminator = EntryBB->getTerminator();
+    BranchInst *Terminator = cast<BranchInst>(EntryBB->getTerminator());
```

Comment:
How do you know that EntryBB ends with a branch?

Reply:
(Incidentally EntryBB is a bad name for this since it sounds like the entry block of the whole function.)

Reply:
Added comment + changed var name